include/cbor_decode.h

```diff
@@ -0,0 +1,95 @@
+/*
```

Comment:
I would really like to see this documented. CDDL/CBOR is not that well understood. 

Doesn't need to be super-detailed, only show the difference between core serialization types and some interesting tidbits like "Are strings null-terminated"?

Reply:
I added a bunch of docs to the header file. Take a look if it covers what you want.

Reply:
Additions accepted!